sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	port, ok := inst["port"]
+	if !ok {
+		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+	}
+	return int(port.(float64))
+}
+
+func GetDbTcpAddr(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	hostname := GetDbHostName(db_name, ns)
+	port := GetDbPort(db_name, ns)
+	return hostname + ":" + strconv.Itoa(port)
+}
+
+func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
+	data, err := io.ReadFile(SONIC_DB_GLOBAL_CONFIG_FILE)
+	if err == nil {
+		//Ref:https://stackoverflow.com/questions/18537257/how-to-get-the-directory-of-the-currently-running-file
+		dir, err := filepath.Abs(filepath.Dir(SONIC_DB_GLOBAL_CONFIG_FILE))
+		if err != nil {
+			panic(err)
+		}
+		var sonic_db_global_config = make(map[string]interface{})
+		err = json.Unmarshal([]byte(data), &sonic_db_global_config)
+		if err != nil {
+			panic(err)
+		}
+		var db_include_file string
+		for _, entry := range sonic_db_global_config["INCLUDES"].([]interface{}) {
+			ns, ok := entry.(map[string]interface{})["namespace"]
+			if !ok {
+				ns = SONIC_DEFAULT_NAMESPACE
+			}
+			_, ok = ns_to_cfgfile_map[ns.(string)]
+			if ok {
+				panic(fmt.Errorf("Global Database config file is not valid(multiple include for same namespace!"))
+			}
+			//Ref:https://www.geeksforgeeks.org/filepath-join-function-in-golang-with-examples/
+			db_include_file = filepath.Join(dir, entry.(map[string]interface{})["include"].(string))
+			ns_to_cfgfile_map[ns.(string)] = db_include_file
+		}
+		if len(ns_to_cfgfile_map) > 1 {
+			sonic_db_multi_namespace = true
+		} else {
+			sonic_db_multi_namespace = false
+		}
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		// Ref: https://stackoverflow.com/questions/23452157/how-do-i-check-for-specific-types-of-error-among-those-returned-by-ioutil-readfi
+		ns_to_cfgfile_map[SONIC_DEFAULT_NAMESPACE] = SONIC_DB_CONFIG_FILE
+		sonic_db_multi_namespace = false
+	} else {
+		panic(err)
+	}
 }
 
 func DbInit() {
-    if sonic_db_init {
-        return
-    }
-    data, err := io.ReadFile(SONIC_DB_CONFIG_FILE)
-    if err != nil {
-        panic(err)
-    } else {
-        err = json.Unmarshal([]byte(data), &sonic_db_config)
-        if err != nil {
-            panic(err)
-        }
-        sonic_db_init = true
-    }
-}
-
-func init() {
-    sonic_db_init = false
+	if sonic_db_init {
+		return
+	}
+	ns_to_cfgfile_map := make(map[string]string)
+	// Ref: https://stackoverflow.com/questions/14928826/passing-pointers-to-maps-in-golang
+	DbGetNamespaceAndConfigFile(ns_to_cfgfile_map)
+	for ns, db_cfg_file := range ns_to_cfgfile_map {
+		data, err := io.ReadFile(db_cfg_file)
+		if err != nil {
+			panic(err)
+		}
+		db_config := make(map[string]interface{})
+		err = json.Unmarshal([]byte(data), &db_config)
+		if err != nil {
+			panic(err)
+		}
+		sonic_db_config[ns] = db_config
+	}
+	sonic_db_init = true
+}
+
+func Init() {
```

Comment:
why do the package init function replaced with the custom Init? 
And this function doesn't do actual initialization, it just set var to false, why do we need to call this function, probably it's better to initialize `sonic_db_init ` during declaration.
`var sonic_db_init  bool = false`

Reply:
This was needed to initialize DB again as part of unit-test so that we can start testing multi-asic once single asic is done.
For example in function `TestGnmiGetMultiNs()` we call `Init()`. Since this make sonic_db_init  as false we will re-initialize the DB again.

Use of init() was existing design so not changing that.

Reply:
So this function doesn't do actual initialization, it does `reset` instead, and so probably it's better to give it appropriate name.
We do not have any reasons to leave `sonic_db_init ` uninitialized, and it can be initialized during declaration.